unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PlayerInfo/PassportPlayerInfoComponentController.cs

```diff
@@ -238,10 +243,16 @@ private async UniTaskVoid AcceptFriendRequestAsync(CancellationToken cancellatio
         private void BlockUser()
         {
             if (ownUserProfile.IsBlocked(currentPlayerId)) return;
-            ownUserProfile.Block(currentPlayerId);
-            view.SetIsBlocked(true);
-            passportApiBridge.SendBlockPlayer(currentPlayerId);
-            socialAnalytics.SendPlayerBlocked(friendsController.IsFriend(currentPlayerId), PlayerActionSource.Passport);
+
+            dataStore.notifications.ConfirmationPopup.Set(ConfirmationPopupData.CreateBlockUserData(
```

Comment:
We have a whole notification system that allows customization, it's implemented as a queue and all the fancy stuff. Cannot we use it instead?

Reply:
I think that the current notification system does not scale properly when you need new properties on a new type of notification and neither complies correctly with MVC/service/bridges architecture. In comparison, im adding properties for cancel button text and the cancel callback, and i don't need all the exposed properties in the notification's model. To support more kind of properties, the notification system model would have to be converted in some kind of god object, which i consider a smell pattern.
I think it is better to have specific type of notification implementations instead of a broad generic one. What do you think?